Fix inconsistent ordering of support base outline and fill

kisslorand opened the pull request:
## Description

This PR fixes an issue in support base generation where the **ordering between outline and fill was not consistently enforced.**

The intended behavior is for each support base to:

**1.** Generate and print its outline first
**2.** Immediately follow with its base fill

In practice, ordering could vary. In some cases the fill was generated before the outline for the same support base. In others, the outline of a support base was generated, followed by base geometry of other supports, with the fill added only later.

## What this PR changes
- Enforces a strict per-support ordering: outline → fill
- Prevents both order inversion (fill before outline) and cross-support interleaving
- Does not alter geometry, extrusion parameters, or visual output

## Scope
- Affects support base path ordering only
- No impact on non-support features

## Result
Support bases are now generated deterministically with outline followed immediately by fill.

Fixes #10711

### Screenshots

- **Before:**
<img width="954" height="697" alt="image" src="https://github.com/user-attachments/assets/e0be37b3-8ea5-4d26-af28-5b0e0bdf7ec0" />

<br><br>

- **After:**
<img width="890" height="553" alt="image" src="https://github.com/user-attachments/assets/a6bb45b9-ef3a-4e01-95d5-d17500c6e2f5" />

kisslorand committed:
Preserve support base outline/fill order

Honor no_sort when emitting support toolpaths to keep outline-first order.
Group tree support base paths (including lightning) into per-area no_sort collections to prevent interleaving across islands.
Keep lightning layer lookup side-effect free.

igiannakas commented:
Perfect, that looks good! Help me understand the rationale if you can, other than being consistent which is absolutely a good idea anyway!

kisslorand committed:
Tag Orca specific changes

Tag Orca specific changes vs. Bambu using the comment //ORCA: . This helps when reviewing merge commits from upstream Bambu so we don't end up causing regressions when pulling in commits from upstream

kisslorand commented:
> Perfect, that looks good! Help me understand the rationale if you can, other than being consistent which is absolutely a good idea anyway!

The outline → fill order for support bases is already generated correctly.

The issue was that this intent could be lost later, allowing outline and fill to be separated or reordered.

Outline and fill belong to the same support base and are meant to be handled as a single unit. This PR doesn’t introduce any new concept; it simply ensures that the existing one is preserved when paths are emitted.

> Also as per the other PR, Could you do me a favour please and tag any orca specific changes vs. bambu using the comment //ORCA:

Done. Please feel free to add suggestions if I missed some.

kisslorand commented:
@SoftFever 
I remember you also worked on this, so I think it might interest you...

SoftFever reviewed (approved):
LGTM
Thank you!

SoftFever commented:
> @SoftFever I remember you also worked on this, so I think it might interest you...

Yes, I tried fixing it earlier, but couldn't get it completely sorted...
Thank you!
